PMM-12409-redesign-service-manage-labels (#1145)
percona/pmm-doc: docs/get-started/roles-and-permissions/lbac.md

## Adding labels

In PMM, labels enable you to identify, categorize, and organize your monitored services and resources. Labels also simplify filtering, grouping, and analyzing data. 

> While adding a service for monitoring in PMM, you can add custom or standard labels. The following labels are available.

Reviewer: WRT `The following labels are available.`: it seems like it was meant to list all the possible labels, right? If not, let's remove it.

Author: yes.

Author: Removed.

**Using PMM UI**

You can set the labels using the User interface as follows:

1. From the **Main** menu, navigate to <i class="uil uil-cog"></i> **Configuration → Add Service**.

2. Select the service you want to add to PMM for monitoring. The page to add the service opens.

3. Enter the details such as **Hostname, Service name, Port, Username, Password**.

>     ![!](../../_images/PMM_access_control_add_labels_services.png)

Reviewer: I don't see such an image being added )

Author: Fixed.

> 4. (Optional) Add the **Labels**.

Reviewer: Important: this is not an `Add Labels` screen, it is rather "Edit labels".

Author: its the Labels which assign while adding a service.

Author: Update the description.

>     ![!](../../_images/PMM_access_control_add_labels_to_add.png)

Reviewer: I'd rename the image as well to match what it represents

Author: Done.

5. (Optional) Check the **Additional Options** that you want to include, such as:

    - **Skip connection check**: Do not check the connection to the database.

    - **Use TLS for database connections**: Enabling TLS (Transport Layer Security) for database connections is an important security measure to protect the confidentiality and integrity of data transmitted between the PMM Server and the monitored database instances.

    - **Skip TLS certificate and hostname validation**: For certain purposes, like debugging and testing, it may be necessary to bypass TLS certificate and hostname validation.

    - **Table statistics limit**: Do not collect table statistics if the number of tables in a database exceeds this limit (defaults to 1000).

    - **Disable comments parsing**: Filter out comments in log files or data streams to isolate valid data.

>     - **Use performance schema**:Performance Schema vs Slow Query Log (default) for monitoring and diagnosing performance issues in your database.

Reviewer: ```suggestion
    - **Use performance schema**: Use Performance Schema instead of Slow Query Log (default) for monitoring and diagnosing performance issues in your database.
```

## Adding labels

In PMM, labels enable you to identify, categorize, and organize your monitored services and resources. Labels also simplify filtering, grouping, and analyzing data. 

While adding a service for monitoring in PMM, you can add custom or standard labels.

**Using PMM UI**

You can set the labels using the User interface as follows:

1. From the **Main** menu, navigate to <i class="uil uil-cog"></i> **Configuration → Add Service**.

2. Select the service you want to add to PMM for monitoring. The page to add the service opens.

    ![!](../../_images/PMM_access_control_add_services.png)

> 3. Enter the details such as **Hostname, Service name, Port, Username, Password, Max query lenght**.

Reviewer 2: Typo in `length`.

## Adding labels

In PMM, labels enable you to identify, categorize, and organize your monitored services and resources. Labels also simplify filtering, grouping, and analyzing data. 

While adding a service for monitoring in PMM, you can add custom or standard labels.

**Using PMM UI**

You can set the labels using the User interface as follows:

1. From the **Main** menu, navigate to <i class="uil uil-cog"></i> **Configuration → Add Service**.

2. Select the service you want to add to PMM for monitoring. The page to add the service opens.

    ![!](../../_images/PMM_access_control_add_services.png)

3. Enter the details such as **Hostname, Service name, Port, Username, Password, Max query length**.

    ![!](../../_images/PMM_access_control_add_main_details.png)

4. (Optional) Enter the **Labels** for the service being monitored.

    ![!](../../_images/PMM_access_control_add_labels_to_add.png)

>     Starting with PMM 2.40.0, you do not have to explicitly add the cluster label while adding a service to PMM for monitoring. The service name will be automatically assigned to the cluster label.

Reviewer: ```suggestion
    With PMM 2.40.0, we start enforcing the cluster label. If you choose not to provide it explicitly while adding a service to PMM for monitoring, PMM will automatically assign the value of the service name to the cluster label.
```

    - **Table statistics limit**: Do not collect table statistics if the number of tables in a database exceeds this limit (defaults to 1000).

    - **Disable comments parsing**: Filter out comments in log files or data streams to isolate valid data.

    - **Use performance schema**: Use Performance Schema instead of Slow Query Log (default) for monitoring and diagnosing performance issues in your database.
 
 **Using pmm-admin**

 You can also assign labels using [pmm-admin](../../details/commands/pmm-admin.md).

## Editing labels

You can edit the labels as follows:

1. From the **Main** menu, navigate to <i class="uil uil-cog"></i> **Configuration → Inventory**.

2. Click on the three dots next to the service you want to edit labels for.

3. Click **Edit**. The **Edit Service** page opens.

4. Edit the labels as per your requirement and click **Save Changes**. The editing service dialogue box opens.

    ![!](../../_images/PMM_access_edit_labels.png)

5. Click **Confirm and save changes**. You will be taken back to the **Inventory/Services** page.

> ### Effect of editing labels

Reviewer 3: edited labels doesn't affect existing time-series(metrics) and will be affected only for new time-series(metrics).

Author: @BupycHuk So this statement is incorrect:" If labels are used extensively in dashboard configurations, editing labels may require updating existing dashboards to ensure they correctly reference the new labels."

Reviewer 3: it depends on dashboards, if dashboards are based on labels values, then yes, but usually no.

Editing existing labels can impact the following PMM functions:

- **Alerting** 

    Editing labels without updating alerting rules can lead to missed alerts. If an alert rule is based on specific labels that are changed or no longer apply, the alert may not trigger when it should.

>     Update the alert rules promptly after ediitng the labels to avoid any problems.

Reviewer: ```suggestion
    Update the alert rules promptly after editing the labels for a smooth alerting experience.
```
